impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -668,6 +671,7 @@ void ContextVK::InitializeCommonlyUsedShadersIfNeeded() const {
 }
 
 void ContextVK::DisposeThreadLocalCachedResources() {
+  Lock lock(desc_pool_mutex_);
```

Comment:
Minimize the scope where this lock is held:

```
{
  Lock lock(desc_pool_mutex_);

 
 cached_descriptor_pool_.erase(std::this_thread::get_id());
}
command_pool_recycler_->Dispose();
```

Reply:
Done